sympy/solvers/solveset.py

```diff
@@ -232,18 +234,22 @@ def _invert_real(f, g_ys, symbol):
 
         if not base_has_sym:
             rhs = g_ys.args[0]
-            if base > S.Zero:
+            try:
+                if base > S.Zero:
+                    return _invert_real(expo,
+                        imageset(Lambda(n, log(n, base, evaluate=False)), g_ys), symbol)
+                elif base < S.Zero:
+                    from sympy.core.power import integer_log
+                    s, b = integer_log(rhs, base)
+                    if b:
+                        return _invert_real(expo, FiniteSet(s), symbol)
+                elif rhs is S.One:
+                    #special case: 0**x - 1
+                    return (expo, FiniteSet(0))
+                return (expo, S.EmptySet)
+            except TypeError:
```

Comment:
Where do you expect this?

Reply:
For equations having symbols, it turns out that relational comparison of symbols with `0` raises an exception. I have added tests in `test_expo_symbols`.

Reply:
then in that case you should rather check before comparing, instead of raising a TypeError, whose significance  is hard to find out from the code

Reply:
It will catch all kinds of TypeError in that block of code and you don't want that. You should only catch the exact exception only where its known to be expected or check before comparing.

Reply:
I think such comparison should not be made when it comes to sympy. I would rather change 
`base > S.Zero` to `base.is_positive`.